### PR TITLE
Type Shooter: Added long word list, file I/O, fixed city and scrolling behavior, changed styles, added new buttons and stats, balanced difficulties

### PR DESCRIPTION
**Pull request by joerob99**
Huge overhaul for Type Shooter, this may be my last PR for this project. I added in the extensive word list from Type Racer and the file I/O function to go along with it, fixed city image and scrolling issues, changed the styles of in-game text to be more easily recognizable, added changing of color for livesLeft from green to red depending on number of lives left, balanced gameplay for the different difficulties, fixed randSector calculation to better reflect reasonable behavior, added an end game button and time survived stat to the in-game screen and fixed issues associated with all of these new updates.

### CODE REVIEW

File: play/type-shooter/js/app.js [hunk 1]
```diff
@@ -34,18 +47,66 @@ const STARTING_LIVES = 10;
 (function initUI() {
     // show instructions modal
     $("#myModal").modal("show");
+
+    // keep page scrolled to the very top
+    document.querySelector("body").scrollTop = 0;
```
Comment (reeswrites) [inline play/type-shooter/js/app.js]: You can actually just do ```document.body``` https://developer.mozilla.org/en-US/docs/Web/API/Document/body.

File: play/type-shooter/js/app.js [hunk 2]
```diff
@@ -34,18 +47,66 @@ const STARTING_LIVES = 10;
...
+    livesLeft.style.webkitTextFillColor = "rgb(0, " + greenTextColor + ", 0)";
+
+    let poppedBalloons = document.getElementById("balloonsPopped");
+    poppedBalloons.style.fontSize = "25px";
+    poppedBalloons.style.fontWeight = "bold";
+    poppedBalloons.style.webkitTextStrokeWidth = "1px";
+    poppedBalloons.style.webkitTextStrokeColor = "black";
+    poppedBalloons.style.webkitTextFillColor = "rgb(255, 255, 255)";
+
+    let timeSurvived = document.querySelector("#timeSurvived");
```
Comment (reeswrites) [inline play/type-shooter/js/app.js]: I would just do ```document.getElementById()``` in order to make it consistent with the adjacent code blocks.

File: play/type-shooter/js/app.js [hunk 3]
```diff
@@ -34,18 +47,66 @@ const STARTING_LIVES = 10;
...
  * Starts the Type Shooter game loop
  */
-function startGame() {
+async function startGame() {
 
+    // wait to finish reading in the text file
+    let promise = await readTextFile(WORD_LIST);
+
+    // let the user know if an error occurred
+    if (promise == -1) {console.log("Error: Input text file could not be read");}
```
Comment (reeswrites) [inline play/type-shooter/js/app.js]: If you're going to have it on one line with only one statement in the if statement, you should just get rid of the brackets.

File: play/type-shooter/js/app.js [hunk 4]
```diff
@@ -158,9 +230,13 @@ function endGame() {
     document.getElementById("playerScore").innerHTML = playerScore;
 
     // calculate timeSurvived and display it in the form 00:00
-    let timeSurvived = ((timer / 60).toFixed() < 10 ? "0" + (timer / 60).toFixed() : (timer / 60).toFixed());
+    let timeSurvived = (Math.floor(timer / 60) < 10 ? "0" + Math.floor(timer / 60) : Math.floor(timer / 60));
     timeSurvived += ":" + (timer % 60 < 10 ? "0" + timer % 60 : timer % 60);
-    document.getElementById("timeSurvived").innerHTML = timeSurvived;
+
+    let timeStats = document.querySelectorAll("#timeSurvived");
+    timeStats.forEach(timeStat => {
```
Comment (reeswrites) [inline play/type-shooter/js/app.js]: With arrow functions with only one statement, you can make them compact where you get rid of the brackets and semicolon to make it fit all in one line. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Arrow_functions

**Review by reeshuffled — CHANGES_REQUESTED** [review]

**Pull request merged**